buildSrc/src/main/java/org/elasticsearch/gradle/plugin/PluginPropertiesTask.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.gradle.plugin;
+
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.elasticsearch.gradle.Version;
+import org.elasticsearch.gradle.VersionProperties;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.InvalidUserDataException;
+import org.gradle.api.Project;
+import org.gradle.api.file.CopySpec;
+import org.gradle.api.plugins.JavaPluginConvention;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Creates a plugin descriptor.
+ */
+public class PluginPropertiesTask extends DefaultTask {
+
+    private PluginPropertiesExtension extension;
+
+    private File descriptorOutput = new File(getProject().getBuildDir(),
+            "generated-resources/plugin-descriptor.properties");
+
+    public PluginPropertiesTask() {
+        Project project = this.getProject();
+        this.extension = project.getExtensions().create("esplugin", PluginPropertiesExtension.class, project);
+    }
+
+    @TaskAction
+    public void configurePluginPropertiesTask() {
+        Project project = this.getProject();
+
+        File templateFile = new File(project.getBuildDir(), "templates");
+        templateFile.getParentFile().mkdirs();
+
+        project.copy((cp) -> {
```

Comment:
nit: the parens are not necessary around `cp`

Reply:
Sorry! I do not understand the comment :neutral_face:

Reply:
You can remove the parenthesis around cp, ie `(cp) -> {` becomes `cp -> {`